core/src/com/tgf/twf/core/world/task/TaskSystem.java

```diff
@@ -55,16 +55,19 @@ public void update(final Duration delta) {
     }
 
     private void assignTaskToIdleAgents() {
-        final MutableInventory cost = new HashMapInventory();
         final Queue<Task> deadLetterQueue = new LinkedList<>();
 
         while (!idleAgents.isEmpty() && !unassignedTasks.isEmpty()) {
             final Agent idleAgent = idleAgents.poll();
             final Task unassignedTask = unassignedTasks.poll();
             final List<Action> actions = unassignedTask.createActions(idleAgent);
-            cost.clear();
-            collectCost(actions, cost);
-            if (idleAgent.getHome().getRelatedComponent(Storage.class).tryConsumeResources(cost)) {
+            final Storage homeStorage = idleAgent.getHome().getRelatedComponent(Storage.class);
+
+            final MutableInventory cost = new HashMapInventory();
+            final MutableInventory production = new HashMapInventory();
+            if (canCostBePaid(actions, homeStorage, cost) && canProductionBeStored(actions, homeStorage, production)) {
```

Comment:
Can we not call `computCost` and `computeProduction` and store the result in `cost` and `production` then pass these to `canCostBePaid` and `canProductionBeStored` respectively? Having side effects in a method prefixed with `can` isn't ideal

Reply:
We can but I wanted to avoid computing both if not necessary. Will refactor.

Reply:
Addressed